File: Components/WinForms/CS/SpreadsheetMainDemo/SpreadsheetRibbonTutorialControlBase.cs
using DevExpress.XtraBars.Ribbon;
using System;

namespace DevExpress.XtraSpreadsheet.Demos {
    public partial class SpreadsheetRibbonTutorialControlBase : SpreadSheetTutorialControlBase {
        public SpreadsheetRibbonTutorialControlBase() {
            InitializeComponent();
            spreadsheetControl1.Options.Culture = DefaultCulture;

            SubscribeEvents();
        }

        #region Properties
        protected internal virtual RibbonPage SelectedRibbonPage { get { return homeRibbonPage1; } }
        #endregion

        protected internal virtual void SubscribeEvents() {
            spreadsheetControl1.CellEndEdit += spreadsheetControl1_CellEndEditOverride;
            spreadsheetControl1.CellValueChanged += spreadsheetControl1_CellValueChangedOverride;
        }

        protected virtual void spreadsheetControl1_CellValueChangedOverride(object sender, SpreadsheetCellEventArgs e) { }
        protected virtual void spreadsheetControl1_CellEndEditOverride(object sender, SpreadsheetCellValidatingEventArgs e) { }

        protected override void DoShow() {
            base.DoShow();
            spreadsheetControl1.Focus();
        }

        protected override void OnLoad(EventArgs e) {
            base.OnLoad(e);
            RibbonPage selectedPage = SelectedRibbonPage;
            if(selectedPage != null)
                ribbonControl1.SelectedPage = selectedPage;
        }
    }
}

File: Components/WinForms/VB/SpreadsheetMainDemo/SpreadsheetRibbonTutorialControlBase.vb
Imports Microsoft.VisualBasic
Imports DevExpress.XtraBars.Ribbon
Imports System

Namespace DevExpress.XtraSpreadsheet.Demos
	Partial Public Class SpreadsheetRibbonTutorialControlBase
		Inherits SpreadSheetTutorialControlBase
		Public Sub New()
			InitializeComponent()
			spreadsheetControl1.Options.Culture = DefaultCulture

			SubscribeEvents()
		End Sub

		#Region "Properties"
		Protected Friend Overridable ReadOnly Property SelectedRibbonPage() As RibbonPage
			Get
				Return homeRibbonPage1
			End Get
		End Property
		#End Region

		Protected Friend Overridable Sub SubscribeEvents()
			AddHandler spreadsheetControl1.CellEndEdit, AddressOf spreadsheetControl1_CellEndEditOverride
			AddHandler spreadsheetControl1.CellValueChanged, AddressOf spreadsheetControl1_CellValueChangedOverride
		End Sub

		Protected Overridable Sub spreadsheetControl1_CellValueChangedOverride(ByVal sender As Object, ByVal e As SpreadsheetCellEventArgs)
		End Sub
		Protected Overridable Sub spreadsheetControl1_CellEndEditOverride(ByVal sender As Object, ByVal e As SpreadsheetCellValidatingEventArgs)
		End Sub

		Protected Overrides Sub DoShow()
			MyBase.DoShow()
			spreadsheetControl1.Focus()
		End Sub

		Protected Overrides Sub OnLoad(ByVal e As EventArgs)
			MyBase.OnLoad(e)
			Dim selectedPage As RibbonPage = SelectedRibbonPage
			If selectedPage IsNot Nothing Then
				ribbonControl1.SelectedPage = selectedPage
			End If
		End Sub
	End Class
End Namespace
